group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -790,6 +812,12 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
                     groupId, memberId, updatedMember.subscribedTopicRegex());
                 bumpGroupEpoch = true;
             }
+        } else {
+            // A new static member replaces an older one with the same instance id, assignments etc.
+            // We will create a new member subscription record for this new member.
+            if (instanceId != null) {
+                records.add(newMemberSubscriptionRecord(groupId, updatedMember));
+            }
```

Comment:
This does not seem correct because we will write a record whenever the member is not updated and we have an instance id. I think that it would be better to capture the fact that we have a new static member in the condition at L801.

Reply:
This bit of code has changed now.